### PR TITLE
PS-5222 : PS crash "Assertion failure: dict0dict.cc:7451:space_id !=…

### PR DESCRIPTION
**Pull request by satya-bodapati**
… SPACE_UNKNOWN"

           during upgrade from PS5.7.23 to PS8.0.13-3 with --innodb_file_per_table=OFF

Problem:
--------

On debug builds, when 5.7 is bootstrapped with innodb_file_per_table=OFF and then upgraded
to 8.0, upgrade fails

This is because, in 8.0, we detect whether mysql database is encrypted or not based on mysql.plugin
table. We look for tablespace flags of "mysql.plugin" table.

With innodb_file_per_table=OFF, mysql.plugin doesn't have tablespace and we assert space_id must be valid.

Fix:
----
Get space_id of "mysql.plugin" from table object (dict_table_t) and use that to determine if database is
encrypted or not for upgrade

### CODE REVIEW

File: mysql-test/t/percona_dd_upgrade_file_per_table_off.test [hunk 1]
```diff
@@ -0,0 +1,89 @@
+--source include/not_windows.inc
+--source include/no_valgrind_without_big.inc
+--source include/have_util_unzip.inc
+
+#--disable_query_log
```
Comment (laurynas-biveinis) [inline mysql-test/t/percona_dd_upgrade_file_per_table_off.test]: Probably want to uncomment this?

File: storage/innobase/include/dict0dict.h [hunk 2]
```diff
@@ -131,13 +131,11 @@ void dict_table_close_and_drop(
...
 void dict_init(void);
 
-dict_table_t*
-dict_table_open_on_index_id(
-/*==================*/
-        space_index_t      table_id,       /*!< in: table id */
-        bool            dict_locked)    /*!< in: TRUE=data dictionary locked */
-        __attribute__((warn_unused_result));
-
+dict_table_t *dict_table_open_on_index_id(
```
Comment (laurynas-biveinis) [inline storage/innobase/include/dict0dict.h:134]: Spurious or it has to make CircleCI happy?

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]

File: mysql-test/t/percona_dd_upgrade_file_per_table_off.test (hunk 1 above)
Comment (satya-bodapati) [reply mysql-test/t/percona_dd_upgrade_file_per_table_off.test]: ah, yes, uncommented for some testing.

File: storage/innobase/include/dict0dict.h (hunk 2 above)
Comment (satya-bodapati) [reply storage/innobase/include/dict0dict.h:134]: made by git clean filter

Comment (laurynas-biveinis) [reply storage/innobase/include/dict0dict.h:134]: Ok, then proceed by what CircleCI says

**Comment by satya-bodapati** [comment]
addressed review comments

**Review by laurynas-biveinis — APPROVED** [review]

**Pull request merged**